subprojects/core/src/main/java/org/gradle/scripts/ScriptLanguage.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.scripts;
+
+import org.gradle.api.Incubating;
+
+/**
+ * Scripting language provider SPI.
+ *
+ * @since 4.0
+ */
+@Incubating
+public interface ScriptLanguage {
+
+    /**
+     * Returns the file extension for scripts written in this script language.
```

Comment:
How about "Returns the file extension *(not including the leading dot)* for scripts written in this script language."?